app/controllers/api/activities_controller.rb

```diff
@@ -0,0 +1,42 @@
+class Api::ActivitiesController < ApplicationController
+  def create
+    Activity.transaction do
+      params[:activities].each do |activity|
+        @last_activity = Activity.new(activity_params(activity))
+        @last_activity.save!
+      end
+    end
+
+  rescue ActiveRecord::RecordInvalid
+    render json: @last_activity.errors.to_hash
+  end
+
+  def index
+    activities = Activity.where(date: Date.parse(params[:date]).to_time.all_day)
```

Comment:
Do not use `to_time` on Date objects, because they know nothing about the time zone in use.